tx/textual/valuerenderer/coins.go

```diff
@@ -76,7 +85,112 @@ func (vr coinsValueRenderer) Format(ctx context.Context, v protoreflect.Value) (
 	}
 }
 
-func (vr coinsValueRenderer) Parse(_ context.Context, screens []Screen) (protoreflect.Value, error) {
-	// ref: https://github.com/cosmos/cosmos-sdk/issues/13153
-	panic("implement me, see #13153")
+func (vr coinsValueRenderer) Parse(ctx context.Context, screens []Screen) (protoreflect.Value, error) {
+	if len(screens) != 1 {
+		return protoreflect.Value{}, fmt.Errorf("expected single screen: %v", screens)
+	}
+
+	if screens[0].Text == emptyCoins {
+		return protoreflect.ValueOfList(utils.NewGenericList([]*basev1beta1.Coin{})), nil
+	}
+
+	coins := strings.Split(screens[0].Text, ", ")
+	metadatas := make([]*bankv1beta1.Metadata, len(coins))
+
+	var err error
+	for i, coin := range coins {
+		coinArr := strings.Split(coin, " ")
+		if len(coinArr) != 2 {
+			return protoreflect.Value{}, fmt.Errorf("invalid coin %s", coins)
+		}
+		metadatas[i], err = vr.coinMetadataQuerier(ctx, coinArr[1])
+		if err != nil {
+			return protoreflect.Value{}, err
+		}
+	}
+
+	parsed, err := parseCoins(coins, metadatas)
+	if err != nil {
+		return protoreflect.Value{}, err
+	}
+
+	if len(parsed) > 1 {
+		return protoreflect.ValueOf(utils.NewGenericList(parsed)), err
+	} else {
+		return protoreflect.ValueOfMessage(parsed[0].ProtoReflect()), err
+	}
```

Comment:
Same as above. Sometimes, we have `repeated Coins` in the proto definition, but only one coin inside. In those case, we want ValueOfList still.

```go
if vr.fd.IsList() {
  return ValueOfList
} else {
  return ValueOfMessage
}
```